static/templates/recent_pm_row.hbs

```diff
@@ -0,0 +1,49 @@
+<tr id="recent_pm:{{pm_key}}" {{#if unread_count}}class="unread_topic"{{/if}} data-unread-count="{{unread_count}}" data-muted="{{muted}}" data-participated="true">
+    <td class="recent_topic_stream">
+        <div class="flex_container">
+            <div class="left_part recent_topics_focusable">
+                <span class="fa fa-envelope"></span>
+                <a href="{{pm_url}}">Private messages</a>
+            </div>
+            <div class="right_part">
+                <span class="stream-privacy filter-icon">
+                    {{#if is_group}}
+                    <span class="fa fa-group"></span>
+                    {{else}}
+                    <span class="{{user_circle_class}} user_circle"></span>
+                    {{/if}}
+                </span>
+            </div>
+        </div>
+    </td>
+    <td class="recent_topic_name">
+        <div class="flex_container">
+            <div class="left_part recent_topics_focusable line_clamp">
+                <a href="{{pm_url}}">{{pm_with}}</a>
+            </div>
+            <div class="right_part">
+                {{#if unread_count}}<span class="unread_count unread_count_pm">{{unread_count}}</span>{{/if}}
+            </div>
+        </div>
+    </td>
+    <td class='recent_topic_users'>
+        <ul class="recent_topics_participants">
+            {{#if other_senders_count}}
+            <li class="recent_topics_participant_item tippy-zulip-tooltip" data-tooltip-template-id="recent_topics_participant_overflow_tooltip:{{pm_key}}">
+                <span class="recent_topics_participant_overflow">+{{other_senders_count}}</span>
+            </li>
+            <template id="recent_topics_participant_overflow_tooltip:{{pm_key}}">{{{other_sender_names_html}}}</template>
+            {{/if}}
+            {{#each senders}}
+                <li class="recent_topics_participant_item tippy-zulip-tooltip" data-tippy-content="{{this.full_name}}">
+                    <img src="{{this.avatar_url_small}}" class="recent_topics_participant_avatar" />
+                </li>
+            {{/each}}
+        </ul>
+    </td>
```

Comment:
What's your thinking on the approach of adding a new template here vs. adding conditionals in `static/templates/recent_topic_row.hbs`?  It seems like several blocks, including this one and the one below, could be identical or nearly identical, and it'd potentially help a lot for readability/maintainability to share code.

Reply:
@timabbott Main concerns for creating new template was introduction of lots of conditionals in `recent_topic_row.hbs` for icons, titles etc. which I think it would clutter the template, and some variables name wouldn't match their use, that could be confusing, or maybe we can use common names such as `thread_key` in place of `topic_key` and `pm_key`.

Reply:
Yeah, I think `conversation_key` would be a good common variable name -- we can do a preparatory refactoring commit to migrates to that.  

I think it's OK for moderately size blocks to be inside conditions -- it's just silly to copy the entire file when large sections are identical. (This often leads to problems where we fix a bug in only one copy).